trie/proof.go

```diff
@@ -12,6 +12,8 @@ import (
 	"github.com/ethereum/go-ethereum/rlp"
 )
 
+type MerkleProof []rlp.RawValue
```

Comment:
I think this type definition doesn't add any value. You can always pass merkle proofs as `[]rlp.RawValue`.
Type definitions make it possible to attach methods to a type, but `MerkleProof` doesn't have any.